main.tf

```diff
@@ -86,7 +86,7 @@ resource "aws_elasticache_replication_group" "default" {
   number_cache_clusters         = var.cluster_mode_enabled ? null : var.cluster_size
   port                          = var.port
   parameter_group_name          = join("", aws_elasticache_parameter_group.default.*.name)
-  availability_zones            = var.cluster_mode_enabled ? null : slice(var.availability_zones, 0, var.cluster_size)
+  availability_zones            = length(var.availability_zones) == 0 ? null : [for n in range(0, var.cluster_size) : element(var.availability_zones, n)]
```

Comment:
Just curious. Is there a difference between **slice(var.availability_zones, 0, var.cluster_size)** and **[for n in range(0, var.cluster_size) : element(var.availability_zones, n)]** ?

Reply:
`element()` has this "wrap-around" behavior built in, so if a list has 3 indexes and you ask for the fourth, it will return the first one for you. That enables you to define - for example - `var.cluster_size = 3` and `var.availability_zones = ["us-east-1a", "us-east-1b"]`, which would fail with `slice()` as there are less indexes (2) than you are asking for (3). And `element()` function returns just one element at a time, thus the `for` loop and `range()` to populate the list.

https://www.terraform.io/docs/language/functions/element.html
https://www.terraform.io/docs/language/functions/slice.html